src/Components/ProductList/index.js

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+import Product from '../Product';
+import { newProductData } from './newProductData';
+import { fetchDelete, fetchGet, fetchPost } from '../../utils/fetches';
+import { getToken } from '../../utils/storage';
+import { API } from '../../config';
+import './index.scss';
+
+class ProductList extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      products: [],
+      page: 1,
+    };
+  }
+
+  //TODO: props로 받은 match와 location에 따라 데이터 요청
+
+  componentDidMount() {
+    //TODO:: '/products?type=new' 데이터 받아오기
+    //const token = getTokenInStorage();
+    // if (token) {
+    //   fetchGet(`/products?type=new&pageSize=10&page=1`, token).then((products) => {
+    //     this.setState({
+    //       products: products.resultList,
+    //     });
+    //   });
+
+    this.setState({
+      products: newProductData.resultList,
+    });
+  }
+
+  componentDidUpdate() {
+    //TODO:: 무한 스크롤을 위한 '/products?type=new' 추가 데이터 받아오기
+    // const token = getTokenInStorage();
+    // const { products, page } = this.state;
+    // if (token) {
+    //   fetchGet(
+    //     `/products?type=new&pageSize=${10}&page=${page + 1}`,
+    //     token,
+    //   ).then((newProducts) => {
+    //     this.setState({
+    //       // 데이터 추가
+    //       products: products.concant(newProducts),
+    //       page: page + 1,
+    //     });
+    //   });
+    // }
+  }
+
+  updatedProducts = (updatedId, type) => {
+    const { products } = this.state;
+
+    const updatedProducts = products.map((product) =>
+      updatedId === product.id
+        ? { ...product, [type]: !product[type] }
+        : product,
+    );
+
+    this.setState({
+      products: updatedProducts,
+    });
+  };
+
+  onToggleLike = (updatedId) => {
+    this.updatedProducts(updatedId, 'like');
+
+    const { products } = this.state;
+
+    products[updatedId - 1].like
+      ? fetchDelete(`http://10.58.7.239:8000/users/like/product/${updatedId}`)
+      : fetchPost(`${API}/users/like/product`, { product_id: updatedId });
+  };
```

Comment:
```suggestion

  if (products[updatedId - 1].like) {
    fetchDelete(`${API}/users/like/product/${updatedId}`)
  } else {
    fetchPost(`${API}/users/like/product`, { product_id: updatedId });
  }
```
논리적으로 코드 진행의 분기가 생겨야 하는 경우 삼항연산자나 && 연산자 보다 if 문을 쓰는 것이 개인적으로는 더 좋다고 생각합니다.
- 조건문의 실행 결과로 떨어지는 값을 사용하는 경우가 아님 (this.handleReset()는 undefined를 리턴)
- 지금은 한 줄이지만 추후에 두줄 이상의 로직이 될 수도 있는데, 그 경우 어차피 if문으로 수정해야 함

Reply:
수정했습니다.